src/coreclr/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -1252,22 +1252,31 @@ void CodeGen::genBaseIntrinsic(GenTreeHWIntrinsic* node)
         }
 
         case NI_Vector128_get_AllBitsSet:
-        case NI_Vector256_get_AllBitsSet:
-        {
             assert(op1 == nullptr);
-            if (varTypeIsFloating(baseType) && compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
+            if (compiler->compOpportunisticallyDependsOn(InstructionSet_AVX))
```

Comment:
Likewise, while the behavior is the same, this is changing the instruction used for floating-point. `ins` was not `pcmpeqd` here but instead was `cmpps` or `cmppd`.

Both the Intel and AMD optimization manuals have notes similar to:
> Previous microarchitectures (before Intel Core microarchitecture) do not have explicit restrictions on
mixing integer and floating-point (FP) operations on XMM registers. For Intel Core microarchitecture,
mixing integer and floating-point operations on the content of an XMM register can degrade performance. Software should avoid mixed-use of integer/FP operation on XMM registers. Specifically:
• Use SIMD integer operations to feed SIMD integer operations. Use PXOR for idiom.
• Use SIMD floating-point operations to feed SIMD floating-point operations. Use XORPS for idiom.
• When floating-point operations are bitwise equivalent, use PS data type instead of PD data type.
  MOVAPS and MOVAPD do the same thing, but MOVAPS takes one less byte to encode the instruction.

and

>  Avoid mixing packed integer operations with packed floating point operations

Which implies that we should try to use the matching floating-point comparisons over the integer comparisons when the target type is floating-point.

Reply:
> Which implies that we should try to use the matching floating-point comparisons over the integer comparisons when the target type is floating-point.

@tannergooding But do we expect the value of `Vector128/256<double>.AllBitsSet` to be used by a floating-point operation in some user scenarios? Do you have some examples in mind?

BTW, have you had a chance to read the SO posts I mentioned?

Reply:
> But do we expect the value of Vector128/256<double>.AllBitsSet

Yes. `AllBitsSet` is a fairly central value for masking, comparing, and blending because it is the value returned when any element is `true`.